### PR TITLE
Ensure that distutils and setuptools._distutils stay consistent

### PR DESCRIPTION
**Pull request by srittau**
Depends on #11757

### CODE REVIEW

File: tests/check_consistent.py [hunk 1]
```diff
@@ -83,6 +83,21 @@ def check_stubs() -> None:
...
+def check_distutils() -> None:
+    """Check whether all setuptools._distutils files are re-exported from distutils."""
+    old_cwd = Path.cwd()
+    try:
+        os.chdir(Path("stubs", "setuptools", "setuptools", "_distutils"))
+        all_setuptools_files = set(Path(".").rglob("*.pyi"))
+        os.chdir(old_cwd / "stubs" / "setuptools" / "distutils")
+        all_distutils_files = set(Path(".").rglob("*.pyi"))
+    finally:
+        os.chdir(old_cwd)
```
Comment (AlexWaygood) [inline tests/check_consistent.py:95]: What about doing something like this? Slightly simpler, and means that we don't need to do the `os.chdir` stuff:

```suggestion
    def all_relative_paths_in_directory(path: str) -> set[Path]:
        return {pyi.relative_to(path) for pyi in Path(path).rglob("*.pyi")}

    all_setuptools_files = all_relative_paths_in_directory("stubs/setuptools/setuptools/_distutils")
    all_distutils_files = all_relative_paths_in_directory("stubs/setuptools/distutils")
```

**Review by AlexWaygood — COMMENTED** [review]
Nice idea!

**Review by AlexWaygood — APPROVED** [review]

**Comment by srittau** [comment]
Random idea for the future: Add an optional "check.py" script to stubs/*/tests to gets called by this script.

* Keeps stuff belonging to each other closer together.
* We could limit calling it if the stubs in question were changed.

File: tests/check_consistent.py [hunk 2]
```diff
@@ -83,6 +83,19 @@ def check_stubs() -> None:
...
 
+def check_distutils() -> None:
+    """Check whether all setuptools._distutils files are re-exported from distutils."""
+
+    def all_relative_paths_in_directory(path: Path) -> set[Path]:
+        return {pyi.relative_to(path) for pyi in path.rglob("*.pyi")}
+
+    all_setuptools_files = all_relative_paths_in_directory(Path("stubs", "setuptools", "setuptools", "_distutils"))
+    all_distutils_files = all_relative_paths_in_directory(Path("stubs", "setuptools", "distutils"))
+    extra_files = all_setuptools_files - all_distutils_files
```
Comment (AlexWaygood) [inline tests/check_consistent.py:94]: Optional: we could do this as a meta-test to ensure that this test doesn't become irrelevant, e.g. if we change our directory structure or something in the future:

```suggestion
    assert all_setuptools_files and all_distutils_files, "Looks like this test might be out of date!"
    extra_files = all_setuptools_files - all_distutils_files
```

**Pull request merged**